services/port_manager/src/main/java/com/futurewei/alcor/portmanager/processor/ProcessorManager.java

```diff
@@ -27,41 +27,57 @@
 public class ProcessorManager {
     private static final Logger LOG = LoggerFactory.getLogger(ProcessorManager.class);
 
-    private static List<IProcessor> statelessProcessors = new ArrayList<>();
-    private static Map<Class, IProcessor> allProcessors = new HashMap<>();
-    private static IProcessor headProcessor;
-    private static IProcessor dataPlaneProcessor;
-    private static IProcessor databaseProcessor;
+    private static List<IProcessor> processors = new ArrayList<>();
+    private static Map<Class, IProcessor> processorMap = new HashMap<>();
 
     private void buildProcessChain() {
```

Comment:
@chenpiaoping trying to understand whether this process relies on the new AfterProcessor annotation. How is the order determined?

Reply:
The order is determined in the instanceProcessor() method.

Reply:
Got it. I think our current assumption for the chain is no more than 2, which should be satisficed in most of cases.